### PR TITLE
feat: Rarity and quality filters

### PR DESCRIPTION
**Pull request by ShadowCat117**
filters

### CODE REVIEW

File: common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public Class<Integer> getType() {
+        return Integer.class;
+    }
+
+    @Override
+    public List<String> getAliases() {
+        return List.of("qual");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java]:
```suggestion
        return List.of("qlt");
```

maybe better?

File: common/src/main/java/com/wynntils/services/itemfilter/statproviders/RarityStatProvider.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public Class<String> getType() {
+        return String.class;
+    }
+
+    @Override
+    public List<String> getAliases() {
+        return List.of("rar");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/services/itemfilter/statproviders/RarityStatProvider.java]: Maybe remove this alias? There is no common shorthand for this. Maybe even the other one aswell

**Review by kristofbolyai — APPROVED** [review]
Aliases are weird but otherwise good!

File: common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java (hunk 1 above)
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java]: Imo we should have "tier" as well, I feel like people say t3 or whatever a lot

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java]: Agreed, good idea

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/services/itemfilter/statproviders/QualityStatProvider.java]: That's true I like that better

**Review by ryanzhoudev — APPROVED** [review]

**Pull request merged**